docs/quick-lint-js.schema.config

```diff
@@ -0,0 +1,65 @@
+    "title": "quick-lint-js configuration",
+    "description": "The quick-lint-js CLI and quick-lint-js editor plugins, can be configured using a quick-lint-js.config file.",
+    "$comment": "Refer https://quick-lint-js.com/config for more details.",
+
+    "type": "object",
+
+    "properties": {
+        "globals": {
+            "description": "Global variables which programs can use. It tells quick-lint-js what global variables to assume exist.",
+            "$comment" : "If the global variables you want are from a popular platform or library, you might want to use 'global-groups' instead.",
+            "type": "object",
+
+            "patternProperties": {
+                "^.*$": {
+                    "anyOf": [
+                        {
+                            "description": "If the value is true, then the variable is defined as if the property’s value was {}. If the value is false, then the variable is not defined, even if a group listed in global-groups would otherwise define the variable.",
+                            "type": "boolean"
+                        },
+                        {
+                            "description": "Specifies whether a global variable can be shadowed or can be assigned to.",
+                            "type": "object",
+
+                            "properties": {
+                                "shadowable": {
+                                    "description": "If true or not present, the variable can redefined in the program’s outer-most scope. If false, the variable cannot be redefined in the program’s outer-most scope.",
+                                    "type": "boolean"
+                                },
+                                "writable": {
+                                    "description": "If true or not present, the variable can be assigned to. If false, the variable cannot be assigned to.",
+                                    "type": "boolean"
+                                }    
+                            },
+                            "additionalProperties": false
+                        }
+                    ]
+                }
+            },
+            "additionalProperties": false
+        },
+
+        "global-groups": {
+            "$comment": "Default global group : {browser, bun, ecmascript, jasmine, jest, jquery, node.js, node.js-es, quickjs, web-worker}",
+            "anyOf": [
+                {
+                    "description": "If true or not present, then it’s as if the value was an array of Default group names. If false, then no global variables are defined aside from those listed in the 'globals'.",
+                    "type": "boolean"
+                },
+                {
+                    "description": "Pre-defined categories of global variables which programs can use. Default global group is not included.",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
```

Comment:
To help with auto-complete and to detect typos, could this be an enum? https://json-schema.org/draft/2020-12/json-schema-validation.html#name-enum

Reply:
Will putting enum, make 'literally-anything' as invalid? It may not allow values other than default values. I am also not completely sure on the behavior.

Reply:
> Will putting enum, make 'literally-anything' as invalid?

`"literally-anything"` would be another enum member. It's *exactly* the string `literally-anything`.

> It may not allow values other than default values.

Currently, the only values in this array that are allowed are the following:
```json
"browser",
"bun",
"ecmascript",
"jasmine",
"jest",
"jquery",
"literally-anything",
"node.js",
"node.js-es",
"quickjs",
"typescript",
"web-worker",
```